### PR TITLE
Pub/Sub: Add examples for schema

### PR DESCRIPTION
**Pull request by kamalaboulhosn**

### CODE REVIEW

File: pubsub/api/Pubsub.Samples.Tests/PubsubFixture.cs [hunk 1]
```diff
@@ -24,7 +24,10 @@ public class PubsubFixture : IDisposable, ICollectionFixture<PubsubFixture>
     public string ProjectId { get; }
     public List<string> TempTopicIds { get; } = new List<string>();
     public List<string> TempSubscriptionIds { get; } = new List<string>();
+    public List<string> TempSchemaIds { get; } = new List<string>();
```
Comment (amanda-tarafa) [inline pubsub/api/Pubsub.Samples.Tests/PubsubFixture.cs:27]: I think you forgot to delete these on Dispose.

File: pubsub/api/Pubsub.Samples/PullMessagesWithFlowControlAsync.cs [hunk 2]
```diff
@@ -39,7 +39,7 @@ public async Task<int> PullMessagesWithFlowControlAsync(string projectId, string
         // threads to maximize throughput.
         Task startTask = subscriber.StartAsync((PubsubMessage message, CancellationToken cancel) =>
         {
-            string text = Encoding.UTF8.GetString(message.Data.ToArray());
+            string text = System.Text.Encoding.UTF8.GetString(message.Data.ToArray());
```
[the remark this answers is not in the document]
Comment (amanda-tarafa) [reply pubsub/api/Pubsub.Samples/PullMessagesWithFlowControlAsync.cs:42]: Ah, we have an Encoding type now on Pub/Sub, never mind :).

File: pubsub/api/Pubsub.Samples.Tests/PullProtoMessagesAsyncTest.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        var result = await _pullProtoMessagesAsyncSample.PullProtoMessagesAsync(_pubsubFixture.ProjectId, subscriptionId, true);
+        Assert.Equal(1, result);
+
+        //Pull the Message to confirm it's gone after it's acknowledged
+        result = await _pullProtoMessagesAsyncSample.PullProtoMessagesAsync(_pubsubFixture.ProjectId, subscriptionId, true);
+        Assert.True(result == 0);
+    }
+
+    [Fact]
+    public async void PullProtoJsonMessagesAsync()
```
Comment (amanda-tarafa) [inline pubsub/api/Pubsub.Samples.Tests/PullProtoMessagesAsyncTest.cs:56]: Thanks for adding tests for all cases.

File: pubsub/api/Pubsub.Samples/CreateAvroSchema.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+using Grpc.Core;
+using System;
+
+public class CreateAvroSchemaSample
+{
+    public Schema CreateAvroSchema(string projectId, string schemaId, string pathToDefinition)
+    {
+        SchemaServiceClient schemaService = SchemaServiceClient.Create();
+        var schemaName = SchemaName.FromProjectSchema(projectId, schemaId);
+        string schemaDefinition = System.IO.File.ReadAllText(pathToDefinition);
```
Comment (amanda-tarafa) [inline pubsub/api/Pubsub.Samples/CreateAvroSchema.cs:27]: Unless if needed for disambiguation, we preffer `using System.IO` and then `File.Read...` here.

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
Only change needed is that you forgot to delete the schemas on Dispose. And a minor nit.
Else, looks really good!

File: pubsub/api/Pubsub.Samples.Tests/PubsubFixture.cs (hunk 1 above)
Comment (kamalaboulhosn) [reply pubsub/api/Pubsub.Samples.Tests/PubsubFixture.cs:27]: Schemas are forever! No, just kidding. Done.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Comment by amanda-tarafa** [comment]
Failures are unrelated and already being tracked. Force merging.

**Pull request merged**